packaging/os/yum.py

```diff
@@ -646,60 +698,40 @@ def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
                 if is_installed(module, repoq, spec, conf_file, en_repos=en_repos, dis_repos=dis_repos):
                     found = True
                     res['results'].append('package providing %s is already installed' % (spec))
-                    
+
             if found:
                 continue
 
+
+            # downgrade - the yum install command will only install or upgrade to a spec version, it will
+            # not install an older version of an RPM even if specified by the install spec. So we need to
+            # determine if this is a downgrade, and then use the yum downgrade command to install the RPM.
+
+            # Get the NEVRA of the requested package
+            (name, ver, rel, epoch, arch) = splitFilename(pkglist.pop())
+
+            # Check if any version of the requested package is installed
+            inst_pkgs = is_installed(module, repoq, name, conf_file, en_repos=en_repos, dis_repos=dis_repos, is_pkg=True)
+            if inst_pkgs:
```

Comment:
It seems like this code may never be hit.  We already called is_installed() before.  If that showed as installed we go to the next iteration of the loop.  This looks like it may be checking the same name with is_installed and therefore it can only be False at this point.  What am I missing?